cdap-hbase-compat-1.1/src/main/java/co/cask/cdap/data2/replicationtool/hbase11/LastWriteTime.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.data2.replicationtool.hbase11;
+
+import co.cask.cdap.common.conf.Constants;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.hadoop.hbase.CoprocessorEnvironment;
+import org.apache.hadoop.hbase.HRegionInfo;
+import org.apache.hadoop.hbase.TableName;
+import org.apache.hadoop.hbase.client.HTableInterface;
+import org.apache.hadoop.hbase.coprocessor.BaseWALObserver;
+import org.apache.hadoop.hbase.coprocessor.ObserverContext;
+import org.apache.hadoop.hbase.coprocessor.WALCoprocessorEnvironment;
+import org.apache.hadoop.hbase.regionserver.wal.WALEdit;
+import org.apache.hadoop.hbase.wal.WALKey;
+
+import java.io.IOException;
+
+/**
+ * HBase coprocessor that tracks WAL writes for all tables to track replication status.
+ * For each region the writeTime of the last WAL entry is written to the REPLICATION_STATE table.
+*/
+public class LastWriteTime extends BaseWALObserver {
+  private TableUpdater tableUpdater = null;
+  private static final Log LOG = LogFactory.getLog(LastWriteTime.class);
+
+  @Override
+  public void start(CoprocessorEnvironment env) throws IOException {
+    LOG.info("LastWriteTime Start received.");
+    String clusterID = env.getConfiguration().get(Constants.ReplicationStatusTool.CLUSTER_ID);
+    if (clusterID == null) {
+      LOG.error(Constants.ReplicationStatusTool.CLUSTER_ID + " not provided in configuration.");
+    }
+    tableUpdater = new TableUpdater(clusterID, Constants.ReplicationStatusTool.WRITE_TIME, env);
+    String tableName = tableUpdater.getReplicationStateTableName(env);
+    HTableInterface htableInterface = env.getTable(TableName.valueOf(tableName));
+    tableUpdater.addHTableInterface(htableInterface);
+  }
+
+  @Override
+  public void postWALWrite(ObserverContext<? extends WALCoprocessorEnvironment> ctx, HRegionInfo info,
+                           WALKey logKey, WALEdit logEdit) throws IOException {
+    if (logKey.getScopes() == null || logKey.getScopes().size() == 0) {
+      //if replication scope is not set for this entry, do not update write time.
+      // This is to save us from cases where some HBase tables do not have replication enabled.
+      // Ideally, you would check scopes against REPLICATION_SCOPE_LOCAL, but cell.getFamily() is expensive.
```

Comment:
HConstants.REPLICATION_SCOPE_LOCAL and HConstants.REPLICATION_SCOPE_GLOBAL.
From the code at ScopeWALEntryFilter looks like we would have to check for each cell in the WALEdit. Also cell.getFamily() documentation says its expensive and should be avoided. 
This check saves us from common cases, but the corner case exists.